[ripgrep] Unable to build

```
error[E0433]: failed to resolve. `crate` can only be used in absolute paths
error[E0433]: failed to resolve. `crate` can only be used in absolute paths
error[E0658]: `crate` in paths is experimental (see issue #45477)
error[E0658]: `crate` in paths is experimental (see issue #45477)
error[E0658]: `crate` in paths is experimental (see issue #45477)
error: aborting due to 5 previous errors
error: Could not compile `same-file`.
error[E0658]: underscore lifetimes are unstable (see issue #44524)
error[E0658]: underscore lifetimes are unstable (see issue #44524)
error[E0658]: underscore lifetimes are unstable (see issue #44524)
error[E0658]: underscore lifetimes are unstable (see issue #44524)
error: aborting due to 4 previous errors
error: Could not compile `termcolor`.
error: build failed
```
Test fails on x86_64:
```
test walk::tests::symlinks ... FAILED

failures:

---- walk::tests::symlinks stdout ----
thread 'walk::tests::symlinks' panicked at 'assertion failed: `(left == right)`
  left: `["a", "a/b", "a/b/foo"]`,
 right: `["a", "a/b", "a/b/foo", "z", "z/foo"]`: single threaded', crates/ignore/src/walk.rs:1880:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    walk::tests::symlinks

test result: FAILED. 130 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out

error: test failed, to rerun pass '-p ignore --lib'
ripgrep failed to build: `cargo test --all` exited with 101
```

--- a/packages/ripgrep.rb
+++ b/packages/ripgrep.rb
@@ -3,9 +3,9 @@ require 'package'
 class Ripgrep < Package
   description 'ripgrep recursively searches directories for a regex pattern'
   homepage 'https://github.com/BurntSushi/ripgrep'
-  version '12.0.1'
-  source_url 'https://github.com/BurntSushi/ripgrep/archive/12.0.1.tar.gz'
-  source_sha256 '5be34aa77a36ac9d8f1297a0d97069e4653e03f61c67d192cee32944cd2b6329'
+  version '12.1.0'
+  source_url 'https://github.com/BurntSushi/ripgrep/archive/12.1.0.tar.gz'
+  source_sha256 'ca2d11dd7b7346734d47ad8073468e9c409fbe85842a608d372b8d4fb36be291'
 
   binary_url ({
   })
@@ -16,12 +16,9 @@ class Ripgrep < Package
 
   def self.build
     case ARCH
-    when 'aarch64'
-      system 'rustup toolchain install 1.25.0-aarch64-unknown-linux-gnu'
-      system 'rustup default 1.25.0-aarch64-unknown-linux-gnu'
-    when 'armv7l'
-      system 'rustup toolchain install 1.25.0-armv7-unknown-linux-gnueabihf'
-      system 'rustup default 1.25.0-armv7-unknown-linux-gnueabihf'
+    when 'aarch64', 'armv7l'
+      system 'rustup toolchain install 1.43.1-armv7-unknown-linux-gnueabihf'
+      system 'rustup default 1.43.1-armv7-unknown-linux-gnueabihf'
     else
       system 'rustup toolchain install stable'
       system 'rustup default stable'

